typed-racket-lib/typed-racket/infer/infer-unit.rkt

```diff
@@ -524,17 +525,22 @@
         [(s (? Mu? t)) (cg s (unfold t))]
 
         ;; find *an* element of elems which can be made a subtype of T
-        [((Intersection: ts) T)
+        [((Intersection: ts raw-prop) T)
          (cset-join
           (for*/list ([t (in-list ts)]
                       [v (in-value (cg t T))]
                       #:when v)
             v))]
         
         ;; constrain S to be below *each* element of elems, and then combine the constraints
```

Comment:
currently it's ignored (which is sound -- we're just generalizing the type a little). I suppose a simple and slightly better thing to do would be extend the lexical environment w/ the proposition so it's in scope for the recursive calls where it might be useful.